### PR TITLE
[11.0][MIG] mass mailing

### PR DESCRIPTION
**Pull request by Benniphx**
Migration done

### CODE REVIEW

**Comment by Benniphx** [comment]
I squashed with my main account cla problem should not be a problem

File: addons/mass_mailing/migrations/11.0.2.0/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            INSERT INTO mail_mass_mailing_contact_list_rel (contact_id,
+            list_id) values(%s,%s)
+            """, (r[0], r[1],))
+
+
+def map_mailing_model_to_mailing_model_id(cr):
+    """Set the mailing_model_id for the mailing_model in mail.mass_mailing"""
+    cr.execute(
+        """
+        SELECT mm.id, m.id FROM mail_mass_mailing AS mm
```
Comment (StefanRijnhart) [inline addons/mass_mailing/migrations/11.0.2.0/post-migration.py]: Try this in a single query using something like `UPDATE mail_mass_mailing mm SET mailing_model_id = im.id FROM ir_model WHERE ir.model = mm.mailing_model` (untested)

Comment (StefanRijnhart) [reply addons/mass_mailing/migrations/11.0.2.0/post-migration.py]: In this case, performance can really take a hit if you don't refactor this into a single query. Or you can use the API https://doc.therp.nl/openupgrade/API.html#openupgradelib.openupgrade.m2o_to_x2m

**Comment by StefanRijnhart** [comment]
Please take the noupdate changes in the ir.cron record.

**Review by StefanRijnhart — CHANGES_REQUESTED** [review]
See my individual comments

File: addons/mass_mailing/migrations/11.0.2.0/post-migration.py (hunk 1 above)
Comment (pedrobaeza) [reply addons/mass_mailing/migrations/11.0.2.0/post-migration.py]: First query is correct. The openupgradelib method can't be used as this was previously a Char field.

File: addons/mass_mailing/migrations/11.0.2.0/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# © 2017 bloopark systems (<http://bloopark.de>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def map_list_id_to_list_ids(cr):
+    """Set the list_ids for the list_id in mail.mass_mailing.contact"""
+    cr.execute(
```
Comment (pedrobaeza) [inline addons/mass_mailing/migrations/11.0.2.0/post-migration.py]: This is the case where you should use openupgradelib `m2o_to_x2m` (https://doc.therp.nl/openupgrade/API.html#openupgradelib.openupgrade.m2o_to_x2m)

File: addons/mass_mailing/migrations/11.0.2.0/post-migration.py (hunk 1 above)
Comment (StefanRijnhart) [reply addons/mass_mailing/migrations/11.0.2.0/post-migration.py]: @pedrobaeza thanks for the correction. Hit the wrong line with that second comment.

**Comment by pedrobaeza** [comment]
I have changed the script for using m2o_to_x2m, and use only one query for the update, so @StefanRijnhart's comment has been honored, so this can be merged.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**